core/java/iesi-core/src/test/java/io/metadew/iesi/script/action/http/HttpExecuteRequestTest.java

```diff
@@ -0,0 +1,453 @@
+
+        when(actionExecution.getAction())
+                .thenReturn(action);
+
+        mockResolvedValues(requestActionParameter);
+
+        HttpComponent httpComponent = createBaseComponent(new ArrayList<>(), new ArrayList<>());
+        mockGetAndTraceHttpComponent(httpComponent);
+
+        HttpExecuteRequest httpExecuteRequest = new HttpExecuteRequest(executionControl, scriptExecution, actionExecution);
+        httpExecuteRequest.prepare();
+
+
+        assertThat(httpExecuteRequest.getExpectedStatusCodes()).isEmpty();
+    }
+
+    @Test
+    void prepareStatusCode() throws HttpRequestBuilderException, URISyntaxException {
+
+        ActionParameter requestActionParameter = createActionParameter("request", "request");
+        ActionParameter statusCodeActionParameter = createActionParameter("expectedStatusCodes", "200");
+        Action action = createAction(requestActionParameter, statusCodeActionParameter);
+        List<String> expectedStatusCode = Collections.singletonList("200");
+
+        when(actionExecution.getAction())
+                .thenReturn(action);
+
+        mockResolvedValues(requestActionParameter, statusCodeActionParameter);
+
+        HttpComponent httpComponent = createBaseComponent(new ArrayList<>(), new ArrayList<>());
+        mockGetAndTraceHttpComponent(httpComponent);
+
+        HttpExecuteRequest httpExecuteRequest = new HttpExecuteRequest(executionControl, scriptExecution, actionExecution);
+        httpExecuteRequest.prepare();
+
+        assertThat(httpExecuteRequest.getExpectedStatusCodes()).isNotEmpty();
+        assertThat(httpExecuteRequest.getExpectedStatusCodes()).get().isEqualTo(expectedStatusCode);
+    }
+
+    @Test
+    void prepareMultipleStatusCode() throws HttpRequestBuilderException, URISyntaxException {
+
+        ActionParameter requestActionParameter = createActionParameter("request", "request");
+        ActionParameter statusCodeActionParameter = createActionParameter("expectedStatusCodes", "200,400,500");
+        Action action = createAction(requestActionParameter, statusCodeActionParameter);
+        List<String> expectedStatusCode = new ArrayList<>(Arrays.asList("200", "400", "500"));
+
+        when(actionExecution.getAction())
+                .thenReturn(action);
+
+        mockResolvedValues(requestActionParameter, statusCodeActionParameter);
+
+        HttpComponent httpComponent = createBaseComponent(new ArrayList<>(), new ArrayList<>());
+        mockGetAndTraceHttpComponent(httpComponent);
+
+        HttpExecuteRequest httpExecuteRequest = new HttpExecuteRequest(executionControl, scriptExecution, actionExecution);
+        httpExecuteRequest.prepare();
+
+        assertThat(httpExecuteRequest.getExpectedStatusCodes()).isNotEmpty();
+        assertThat(httpExecuteRequest.getExpectedStatusCodes()).get().isEqualTo(expectedStatusCode);
+    }
+
+    @Test
+    void prepareNoProxyConnection() throws HttpRequestBuilderException, URISyntaxException {
```

Comment:
test this in the first unit test, the 'use case' is the same: only request defined